in_toto/in_toto_sign.py

```diff
@@ -0,0 +1,265 @@
+      signs the file, and then returns the link file as an object.
+
+    <Arguments>
+      link - path to the key file
+      key - the key to be used for signing
+
+    <Exceptions>
+      None
+
+    <Returns>
+      An object containing the contents of the link file after
+      adding the signature which replaces the old signatures
+    """
+
+    # reading the link file and making an object
+    signable_object = link_import.read_from_file(link)
+
+    # import rsa key from the filepath
+    rsa_key = in_toto.util.import_rsa_key_from_file(key)
+
+    # check if the key corresponds to the correct format
+    securesystemslib.formats.KEY_SCHEMA.check_match(rsa_key)
+
+    # core working
+    signature = securesystemslib.keys.create_signature(rsa_key, signable_object.payload)
+    signable_object.signatures = []
+    signable_object.signatures.append(signature)
+
+    return signable_object
+
+
+def verify_sign(link, key_pub):
+    """
+    <Purpose>
+      Verifies the signature field in the link file, given a public key
+
+    <Arguments>
+      link - path to the link file
+      key_pub - public key to be used to verification
+
+    <Exceptions>
+      Raises SignatureVerificationError
+        - 'Invalid Signature' : when the verification fails
+        - 'Signature Key Not Found' : when KeyError occurs
+        - 'No Signatures Found' - when no signature exists
+
+    <Returns>
+      Boolean True when the verification is success
+    """
+
+    signable_object = link_import.read_from_file(link)
+    link_key_dict = in_toto.util.import_rsa_public_keys_from_files_as_dict(
+      [key_pub])
+
+    signable_object.verify_signatures(link_key_dict)
+
+    return True
+
+
+def parse_args():
+    """
+    <Purpose>
+      A function which parses the user supplied arguments.
+
+    <Arguments>
+      None
+
+    <Exceptions>
+      None
+
+    <Returns>
+      Parsed arguments (args object)
+    """
+    parser = argparse.ArgumentParser(
+        description="in-toto-sign : Signs etc etc")
+
+    lpad = (len(parser.prog) + 1) * " "
+
+    parser.usage = ("\n"
+                    "--key <signing key> | <verifying key>\n{0}"
+                    "<sign | verify>\n{0}"
+                    "[--replaceall]\n{0}"
+                    "[--infixkeyid]\n{0}"
+                    "<path/to/signable>"
+                    "[--verbose]\n\n"
+                    .format(lpad))
+
+    in_toto_args = parser.add_argument_group("in-toto-sign options")
+
+    in_toto_args.add_argument("-k", "--key", type=str, required=True,
+                              help="Path to private key to sign link metadata (PEM)")
+
+    in_toto_args.add_argument("operator", type=str, choices=['sign', 'verify'],
+                              help="sign or verify")
+
+    in_toto_args.add_argument("-r", "--replaceall", action="store_true",
+                              help = "Whether to replace all the old signatures"
```

Comment:
Please remove whitespaces before and after assignment operator when using keyword arguments consistently throughout the script.